hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitter.java

```diff
@@ -93,41 +82,46 @@ public class OffsetCommitter implements Runnable {
     private final Map<SubscriptionPartition, Long> maxCommittedOffsets = new HashMap<>();
 
     public OffsetCommitter(
-            OffsetQueue offsetQueue,
             ConsumerPartitionAssignmentState partitionAssignmentState,
-            MessageCommitter messageCommitter,
-            int offsetCommitPeriodSeconds,
             MetricsFacade metrics
     ) {
-        this.offsetQueue = offsetQueue;
         this.partitionAssignmentState = partitionAssignmentState;
-        this.messageCommitter = messageCommitter;
-        this.offsetCommitPeriodSeconds = offsetCommitPeriodSeconds;
         this.obsoleteCounter = metrics.offsetCommits().obsoleteCounter();
         this.committedCounter = metrics.offsetCommits().committedCounter();
         this.timer = metrics.offsetCommits().duration();
     }
 
-    @Override
-    public void run() {
+    public Set<SubscriptionPartitionOffset> calculateOffsetsToBeCommitted(Map<SubscriptionPartitionOffset, MessageState> offsets) {
         try (HermesTimerContext ignored = timer.time()) {
-            // committed offsets need to be drained first so that there is no possibility of new committed offsets
-            // showing up after inflight queue is drained - this would lead to stall in committing offsets
-            ReducingConsumer committedOffsetsReducer = processCommittedOffsets();
+            // committed offsets need to be copied first so that there is no possibility of new committed offsets
```

Comment:
does this comment still applies ? the `offsets` is a snapshot so this cannot be updated during the algorithm working 🤔

Reply:
indeed, I will remove it